Python/Product/Analysis/ModulePath.cs

```diff
@@ -145,8 +142,11 @@ public ModulePath(string fullname, string sourceFile, string libraryPath)
             RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
         private static readonly Regex PythonBinaryRegex = new Regex(@"^(?!\d)(?<name>(\w|_)+)\.((\w|_|-)+?\.)?(pyd|so|dyLib)$",
             RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
-        private static readonly Regex PythonCompiledRegex = new Regex(@"^(?!\d)(?<name>(\w|_)+)\.(((\w|_|-)+?\.)?pyd|py[co]|so|dyLib)$",
-            RegexOptions.CultureInvariant | RegexOptions.IgnoreCase);
+
+        private static bool IsCompiledModule(string name) {
+            var ext = Path.GetExtension(PathUtils.GetFileName(name)).ToLowerInvariant();
+            return CompiledModuleExtensions.Any(x => x == ext);
```

Comment:
Why not use `.EndsWithOrdinal(x, ignoreCase: true)`?

Reply:
OK

Reply:
```csharp
private static bool IsCompiledModule(string name) =>
    CompiledModuleExtensions.Any(ext => name.EndsWithOrdinal(ext, true))
```